lib/puppet/data_providers/hiera_config.rb

```diff
@@ -58,12 +59,20 @@ def self.create_config_type
     # @api public
     def initialize(config_root)
       @config_root = config_root
-      @config_path = config_root + 'hiera.yaml'
-      if @config_path.exist?
-        @config = validate_config(HieraConfig.symkeys_to_string(YAML.load_file(@config_path)))
+      config_path_hocon = config_root + 'hiera.conf'
+      config_path_yaml = config_root + 'hiera.yaml'
+      if config_path_hocon.exist?
+        @config = validate_config(HieraConfig.symkeys_to_string(Hocon.load(File.absolute_path(config_path_hocon))))
```

Comment:
From [these docs](https://github.com/typesafehub/config), `load`returns a `Config` object which is an immutable map.